integrations/terraform-cloud/client.py

```diff
@@ -34,39 +39,72 @@ def __init__(self, terraform_base_url: str, auth_token: str) -> None:
         self.client = http_async_client
         self.client.headers.update(self.base_headers)
 
+        self.rate_limiter = AsyncLimiter(TERRAFORM_RATE_LIMIT, 1)
+        self._remaining = TERRAFORM_RATE_LIMIT
+        self._reset_time = time.time()
+
     async def send_api_request(
         self,
         endpoint: str,
         method: str = "GET",
         query_params: Optional[dict[str, Any]] = None,
         json_data: Optional[dict[str, Any]] = None,
     ) -> dict[str, Any]:
-        logger.info(f"Requesting Terraform Cloud data for endpoint: {endpoint}")
-        try:
-            url = f"{self.api_url}/{endpoint}"
-            logger.info(
-                f"URL: {url}, Method: {method}, Params: {query_params}, Body: {json_data}"
-            )
-            response = await self.client.request(
-                method=method,
-                url=url,
-                params=query_params,
-                json=json_data,
-            )
-            response.raise_for_status()
+        async with self.rate_limiter:
+            try:
+                url = f"{self.api_url}/{endpoint}"
+                response = await self.client.request(
+                    method=method,
+                    url=url,
+                    params=query_params,
+                    json=json_data,
+                )
 
-            logger.info(f"Successfully retrieved data for endpoint: {endpoint}")
+                self._remaining = int(
+                    response.headers.get("x-ratelimit-remaining", TERRAFORM_RATE_LIMIT)
+                )
+                reset_in = float(
+                    response.headers.get("x-ratelimit-reset", 1)
+                )  # Default to 1 second
+                self._reset_time = time.time() + reset_in
```

Comment:
is this needed?

Reply:
I left it in as I was testing performance

Reply:
I am removing it